packages/apps/shopify-api/runtime/http/headers.ts

```diff
@@ -54,41 +96,56 @@ export function addHeader(headers: Headers, key: string, value: string) {
   list.push(value);
 }
 
-function canonicalizeValue(value: any): any {
-  if (typeof value === 'number') return value.toString();
-  return value;
+/**
+ * Canonicalizes a header value, converting numbers to strings.
+ *
+ * @param value - The value to canonicalize.
+ * @returns The canonicalized value as a string.
+ */
+function canonicalizeValue(value: unknown): string {
+  return typeof value === 'number' ? value.toString() : String(value);
 }
 
+/**
+ * Canonicalizes all headers in the headers object by ensuring consistent header names and values.
+ *
+ * @param hdr - The headers object to canonicalize.
+ * @returns The headers object with canonicalized header names and values.
+ */
 export function canonicalizeHeaders(hdr: Headers): Headers {
   for (const [key, values] of Object.entries(hdr)) {
     const canonKey = canonicalizeHeaderName(key);
     if (!hdr[canonKey]) hdr[canonKey] = [];
-    if (!Array.isArray(hdr[canonKey]))
+    if (!Array.isArray(hdr[canonKey])) {
       hdr[canonKey] = [canonicalizeValue(hdr[canonKey])];
+    }
     if (key === canonKey) continue;
     delete hdr[key];
-    (hdr[canonKey] as any).push(
+    (hdr[canonKey] as string[]).push(
       ...[values].flat().map((value) => canonicalizeValue(value)),
     );
   }
   return hdr;
 }
 
-export function removeHeader(headers: Headers, needle: string) {
+/**
+ * Removes a header from the headers object by canonicalizing the header name.
```

Comment:
This sounds a little bit off.
```suggestion
 * Removes a header from the headers object.
```